File: android/rctmgl/src/main/kotlin/com/mapbox/rctmgl/components/styles/layers/RCTMGLBackgroundLayer.java
package com.mapbox.rctmgl.components.styles.layers;

import android.content.Context;

import com.mapbox.maps.extension.style.layers.generated.BackgroundLayer;
import com.mapbox.rctmgl.components.styles.RCTMGLStyle;
import com.mapbox.rctmgl.components.styles.RCTMGLStyleFactory;

public class RCTMGLBackgroundLayer extends RCTLayer<BackgroundLayer> {
    public RCTMGLBackgroundLayer(Context context) {
        super(context);
    }

    @Override
    public BackgroundLayer makeLayer() {
        return new BackgroundLayer(mID);
    }

    @Override
    public void addStyles() {
        RCTMGLStyleFactory.setBackgroundLayerStyle(mLayer, new RCTMGLStyle(getContext(), mReactStyle, mMap));
    }
}

File: android/src/main/java-v10/com/mapbox/rctmgl/components/styles/layers/RCTMGLBackgroundLayer.kt
package com.mapbox.rctmgl.components.styles.layers

import android.content.Context
import com.mapbox.maps.extension.style.layers.generated.BackgroundLayer
import com.mapbox.rctmgl.components.styles.RCTMGLStyle
import com.mapbox.rctmgl.components.styles.RCTMGLStyleFactory
import com.mapbox.rctmgl.utils.Logger

class RCTMGLBackgroundLayer(context: Context?) : RCTLayer<BackgroundLayer?>(
    context!!
) {
    override fun makeLayer(): BackgroundLayer {
        return BackgroundLayer(iD!!)
    }

    override fun addStyles() {
        mLayer?.also {
            RCTMGLStyleFactory.setBackgroundLayerStyle(
                it,
                RCTMGLStyle(context, mReactStyle, mMap!!)
            )
        } ?: run {
            Logger.e("RCTMGLBackgroundLayer", "mLayer is null")
        }
    }
}